src/pages/Admin/User/UserListPage.tsx

```diff
@@ -1,41 +1,18 @@
 import React, { FC } from 'react';
 import { PageProps } from '../..';
 import ListPage from '../../../components/Admin/ListPage';
-import { SearchableListItem } from '../../../components/Admin/SearchableList';
-import { Loading } from '../../../components/core';
-import { useApolloErrorHandler } from '../../../hooks';
-import { useDeleteUserMutation, useUsersQuery } from '../../../hooks/generated/graphql';
+import UserListContainer from '../../../containers/user/user-list/UserListContainer';
 
 interface UserListPageProps extends PageProps {}
 
 export const UserListPage: FC<UserListPageProps> = ({ paths }) => {
-  const { data, loading } = useUsersQuery({ fetchPolicy: 'cache-and-network' });
-
-  const users = data?.users || [];
-
-  const userList = users.map(u => ({ id: u.id, value: `${u.displayName} (${u.email})`, url: `${u.id}/edit` }));
-  const handleError = useApolloErrorHandler();
-
-  const [deleteUser] = useDeleteUserMutation({
-    refetchQueries: ['users'],
-    awaitRefetchQueries: true,
-    onError: handleError,
-  });
-
-  const handleDelete = (item: SearchableListItem) => {
-    deleteUser({
-      variables: {
-        input: {
-          ID: item.id,
-        },
-      },
-    });
-  };
-
-  if (loading) {
-    return <Loading text={'Loading Users ...'} />;
-  }
-  return <ListPage data={userList} paths={paths} onDelete={handleDelete} />;
+  return (
+    <UserListContainer
+      component={({ userList, onDelete, loading }) => (
+        <ListPage data={userList} paths={paths} onDelete={onDelete} loading={loading} />
```

Comment:
Please use children function for that use case.
```
<UserListContainer>
  {({ userList, onDelete, loading }) => (
    <ListPage data={userList} paths={paths} onDelete={onDelete} loading={loading} />
  )}
</UserListContainer>

Reply:
`component=` signature is only for the exact match between the provided props and the child component props.

Reply:
> `component=` signature is only for the exact match between the provided props and the child component props.

Can we forbid using a child function in the component prop?
Maybe just `ComponentClass` instead of `ComponentType`?

Reply:
Well, a component is a function. I think we can take another direction and just pass a component as a child fn:
```
<Container>
  {Component}
</Container>